aws/resource_aws_sagemaker_lifecycle_configuration_test.go

```diff
@@ -0,0 +1,211 @@
+	var lifecycleConfig sagemaker.DescribeNotebookInstanceLifecycleConfigOutput
+	rName := acctest.RandomWithPrefix(SagemakerLifecycleConfigurationResourcePrefix)
+	resourceName := "aws_sagemaker_lifecycle_config.test"
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckAWSSagemakerLifecycleConfigurationDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccSagemakerLifecycleConfigurationConfig_Basic(rName),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckAWSSagemakerLifecycleConfigurationExists(resourceName, &lifecycleConfig),
+
+					resource.TestCheckResourceAttr(resourceName, "name", rName),
+					resource.TestCheckResourceAttr(resourceName, "on_create", base64Encode([]byte("echo foo"))),
+					resource.TestCheckResourceAttr(resourceName, "on_start", base64Encode([]byte("echo bar"))),
+				),
+			},
+			{
+				Config: testAccSagemakerLifecycleConfigurationConfig_Update(rName),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckAWSSagemakerLifecycleConfigurationExists(resourceName, &lifecycleConfig),
+
+					resource.TestCheckResourceAttr(resourceName, "on_create", base64Encode([]byte("echo bla"))),
+					resource.TestCheckResourceAttr(resourceName, "on_start", base64Encode([]byte("echo blub"))),
+				),
+			},
+			{
+				ResourceName:      resourceName,
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+		},
+	})
+}
+
+func testAccCheckAWSSagemakerLifecycleConfigurationExists(resourceName string, lifecycleConfig *sagemaker.DescribeNotebookInstanceLifecycleConfigOutput) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		rs, ok := s.RootModule().Resources[resourceName]
+		if !ok {
+			return fmt.Errorf("not found: %s", resourceName)
+		}
+
+		if rs.Primary.ID == "" {
+			return fmt.Errorf("no ID is set")
+		}
+
+		conn := testAccProvider.Meta().(*AWSClient).sagemakerconn
+		output, err := conn.DescribeNotebookInstanceLifecycleConfig(&sagemaker.DescribeNotebookInstanceLifecycleConfigInput{
+			NotebookInstanceLifecycleConfigName: aws.String(rs.Primary.ID),
+		})
+
+		if err != nil {
+			return err
+		}
+
+		if output == nil {
+			return fmt.Errorf("no SageMaker notebook instance lifecycle configuration")
+		}
+
+		*lifecycleConfig = *output
+
+		return nil
+	}
+}
+
+func testAccCheckAWSSagemakerLifecycleConfigurationDestroy(s *terraform.State) error {
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != "aws_sagemaker_lifecycle_config" {
+			continue
+		}
+
+		conn := testAccProvider.Meta().(*AWSClient).sagemakerconn
+		lifecycleConfig, err := conn.DescribeNotebookInstanceLifecycleConfig(&sagemaker.DescribeNotebookInstanceLifecycleConfigInput{
+			NotebookInstanceLifecycleConfigName: aws.String(rs.Primary.ID),
+		})
+
+		if err != nil {
+			if isAWSErr(err, "ValidationException", "") {
+				return nil
```

Comment:
`CheckDestroy` functions should use `continue` instead of `return nil` to allow checking multiple declarations of the resource.